### DISCUSSION TITLE
Adding Locality-based Zone/Priority label to metrics?

### DISCUSSION THREAD

**Discussion by diranged** [istio/istio#49677] [Q&A]
In an environment where [Locality-based Load Balancing](https://istio.io/latest/docs/tasks/traffic-management/locality-load-balancing/) is being used, it can be critical to understand the decisions that Envoy is making with regards to which zone is being used for a particular request. I'd like to see the various `istio_...` metrics have the ability to include something like a `priority=XX` or `zone=XX` label.

I'm curious if this is something we can add  via https://istio.io/latest/docs/tasks/observability/metrics/classify-metrics/.. Any thoughts on this?

**Comment by howardjohn** [comment]
I think you can add a custom metric with the `upstream_peer.labels['locality'].value`. Note this is just in the `tagOverrides`, you don't need the `istio-attributegen-filter`

Credits to https://github.com/tetratelabs/istio-cost-analyzer/blob/b9135a1c6a13c9808a7e195d8781bb1558d87b24/pkg/kube.go#L316

**Reply by zirain** [reply]
I don't think `upstream_peer.labels['locality'].value` will work by default, that tool will mutate pod labels. 

https://github.com/tetratelabs/istio-cost-analyzer/blob/b9135a1c6a13c9808a7e195d8781bb1558d87b24/mutating-webhook/cmd/mutating-webhook/main.go#L97

**Reply by howardjohn** [reply]
Oh. bummer. It should be in the endpoint metadata, probably it can be extracted -- we extract other things from there. Just not sure the exact expression

**Reply by diranged** [reply]
@howardjohn Does this qualify as something we should ask for in a feature request?

**Comment by howardjohn** [comment]
I am fairly sure you can do this today, just need to find out the
expression to do so.

**Comment by zirain** [comment]
this may give a little help https://istio.slack.com/archives/C382V8Q92/p1709040956569369?thread_ts=1708424603.944049&cid=C382V8Q92

**Reply by diranged** [reply]
Near as I can tell from that thread - there's no current built in answer for how to do this.. or am I missing something?
